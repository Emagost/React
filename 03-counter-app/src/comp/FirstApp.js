import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ 
    hello = 'Hello world',
    subtitle = 'Subtitle'
    }) => {

    return (
        <>
            <h1>{hello}</h1>
            {/* <pre>{JSON.stringify(hello,null,3)}</pre> */}
            <p>{subtitle}</p>
        </>
        );

}

FirstApp.propTypes = {
    hello: PropTypes.string
}

// FirstApp.defaultProps = {
//     subtitle: 'Subtitle'
// }

export default FirstApp;