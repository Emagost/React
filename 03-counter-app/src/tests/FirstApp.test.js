import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme'
import FirstApp from '../comp/FirstApp';

describe('First test <FirstApp />', () => {
    
    // test('should show the msg "Hello Emanuel"', () => {
    //     const hello = 'Hello Emanuel';
    //     const { getByText } = render( <FirstApp hello= { hello }/>);
    //     expect(getByText(hello)).toBeInTheDocument();
    // })

    test('should show the <FirstApp/> correctly', () => {
        
        const hello = 'Hello Emanuel';
        const wrapper = shallow(<FirstApp saludo={hello}/>);
        
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show the subtitle send by props', () => {

        const hello = 'Hello, Emanuel';
        const subtitle = 'Hello, I am subtitle';
        const wrapper = shallow(
            <FirstApp 
                saludo={hello}
                subtitle={subtitle}
            />
        );
        
            const text = wrapper.find('p').text();

            // console.log(subtitle);
            // console.log(text);

            expect (text).toBe (subtitle);

    })
    
    

})
