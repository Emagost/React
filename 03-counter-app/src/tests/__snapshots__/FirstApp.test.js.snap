// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First test <FirstApp /> should show the <FirstApp/> correctly 1`] = `
<Fragment>
  <h1>
    Hello world
  </h1>
  <p>
    Hello Subtitle
  </p>
</Fragment>
`;
